Rooms

Tico39 opened the pull request:
<!---
Pull request template for 603 Game 3. Feel free to remove comments as you fill out information.
Original template by Rob Reddick
--->
## Summarize what is being added
<!--- Can use bullet lists here to cover new additions, but give a bit more detail than what is given in commit messages --->
- Added 5 room prefabs with marked exit points and NavMesh
- Added corridor tile
- Added door prefabs

## Please describe how to test
<!---Give the important steps needed for testing your main changes--->
n/a
Stored in Prefabs -> Rooms

## Relevant Screenshots
<!---Paste in some screenshots to help reduce need of excess testing. If screenshots are not relevant here, then remove this section--->
<img width="932" alt="Screenshot 2024-11-11 at 6 36 49 PM" src="https://github.com/user-attachments/assets/a0485d62-6272-4f89-8c9c-46a63a279a09">

## Issue worked on
<!---Paste in a link to the issue this PR addresses--->
https://timeisup.atlassian.net/browse/SCRUM-94

## What Sprint is this due in?
<!---Only need to indicate here which week/deliverable the PR is for--->
Sprint 4

Tico39 committed:
Added Assets

Tico39 committed:
Added first 2 rooms

started room 3

Tico39 committed:
More rooms

- Added more rooms
- Added corridor tile
- Added doors prefab

Tico39 committed:
Minor corridor adjustment

heenathadani reviewed (commented):
1. Please add a layer to exits "Exit" in a similar way the layer Ground has been added to the Ground Gameobject.
2. I am not able to see the size of the room (I only have size of each tile) (If you remember you showed me the size of the room through the box component in class that day)
3. Is it possible to have block walls inside the rooms to have a different material than the outside of the dungeon?

Tico39 committed:
Addressing Comments

- Added Box Collision Component to determine room size
- Exits are now in the "Exit" layer
- Added NavMesh obstacle components where appropriate

heenathadani committed:
Merge branch 'Sprints/Sprint-4' into Rooms

heenathadani committed:
changed layer of the enemy prefab

Tico39 committed:
Removed unnecessary NavMesh components

RohitK-RIT reviewed (approved):
![image](https://github.com/user-attachments/assets/ad7944b6-0fa1-4d74-977b-2849d9401a98)
